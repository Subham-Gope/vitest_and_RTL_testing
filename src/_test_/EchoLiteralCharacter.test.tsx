// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import EchoLiteralCharacter from "../components/EchoLiteralCharacter";

// describe("Testing the EchoLiteralCharacter.tsx", () => {
//   test("registers literal input and displays it, like eg: {}", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(
//       <EchoLiteralCharacter
//         message={"Testing Component"}
//         assignmentNumber={5}
//         status={true}
//       />
//     );

//     expect(container).toMatchSnapshot();

//     const textArea = screen.getByLabelText(/enter your/i);
//     expect(textArea).toBeInTheDocument();

//     // Initaiting the user Interaction
//     textArea.focus();
//     await user.keyboard("{\\{}}{\\{}}");
//     expect(textArea).toHaveValue("{}{}");
//   });
// });
