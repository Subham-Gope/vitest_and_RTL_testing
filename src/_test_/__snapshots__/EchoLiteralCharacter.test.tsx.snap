// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the EchoLiteralCharacter.tsx > registers literal input and displays it, like eg: {} 1`] = `
<div>
  <div
    class="flex flex-col items-center h-scree w-screen"
  >
    <h1>
      Message: 
      Testing Component
      , Assignment Number: 
      5
      , Completion Status: 
    </h1>
    <div
      class="flex flex-col items-center px-8 py-12"
    >
      <label
        for="literal"
      >
        Enter your Literal text
      </label>
      <textarea
        id="literal"
        placeholder="Enter your text"
      />
    </div>
  </div>
</div>
`;
